sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -53,14 +53,27 @@ object DateTimeUtils {
 
   final val MILLIS_PER_DAY = SECONDS_PER_DAY * 1000L
 
-  // number of days in 400 years
+  // number of days in 400 years by Gregorian calendar
   final val daysIn400Years: Int = 146097
+
+  // In Julian calendar every year which is a multiple of 4 is a leap year.
+  // So in 400 years there are 3 extra leap years compared to the Gregorian calendar:
+  // 100, 200 and 300 (as multiples of 100 but not 400).
```

Comment:
Actually I must read https://en.wikipedia.org/wiki/Julian_calendar#Year_length;_leap_years to understand why there are three extra days here. I'm not sure if this added comments are enough clear to others.

Reply:
+1, some kind of rewording of:
```
The Gregorian calendar has the same months and month lengths as the Julian calendar, but, in the Gregorian calendar, year numbers evenly divisible by 100 are not leap years, except that those evenly divisible by 400 remain leap years.
```
would be great here IMHO